plugins/inputs/directory_monitor/directory_monitor_test.go

```diff
@@ -19,12 +20,8 @@ func TestCSVGZImport(t *testing.T) {
 	testCsvGzFile := "test.csv.gz"
 
 	// Establish process directory and finished directory.
-	finishedDirectory, err := os.MkdirTemp("", "finished")
-	require.NoError(t, err)
-	processDirectory, err := os.MkdirTemp("", "test")
-	require.NoError(t, err)
-	defer os.RemoveAll(processDirectory)
-	defer os.RemoveAll(finishedDirectory)
+	finishedDirectory := t.TempDir()
+	processDirectory := t.TempDir()
```

Comment:
I think this plugin needs the directory and finished directory to be different. The plugin looks for files in processDirectory and puts files in finishedDirectory. If they're the same, it could use its own output files as input files and get stuck in a loop.

Same goes for the five other places later in this file that follow the same pattern

Reply:
The first call of `t.TempDir()` returns `TestCSVGZImport/001`, the second call returns `TestCSVGZImport/002` etc. `t.TempDir()` uses a single parent directory (the test name) for all the temporary directories, each numbered sequentially ([source code](https://github.com/golang/go/blob/91b9915d3f6f8cd2e9e9fda63f67772803adfa03/src/testing/testing.go#L1057))

Therefore, `finishedDirectory` will be `TestCSVGZImport/001` and `processDirectory` will be `TestCSVGZImport/002`.

Reply:
Oops, I didn't notice that TempDir's doc string says "Each subsequent call to t.TempDir returns a unique directory". Thanks for setting me straight.